android-sdk/src/main/java/org/sagebionetworks/bridge/android/data/Archive.java

```diff
@@ -0,0 +1,121 @@
+package org.sagebionetworks.bridge.android.data;
+
+import com.google.common.collect.Lists;
+
+import org.joda.time.DateTime;
+import org.sagebionetworks.bridge.android.BridgeConfig;
+import org.sagebionetworks.bridge.rest.RestUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Created by jyliu on 1/30/2017.
+ */
+
+public class Archive {
+    private static final String ARCHIVE_INFO_FILE_NAME = "info.json";
+
+    private final List<ArchiveFile> dataFiles;
+    private final ArchiveInfo archiveInfo;
+
+    private Archive(List<ArchiveFile> dataFiles, ArchiveInfo archiveInfo) {
+        this.dataFiles = dataFiles;
+        this.archiveInfo = archiveInfo;
+    }
+
+    public ZipOutputStream writeTo(OutputStream os) throws IOException {
+        ZipOutputStream zos = new ZipOutputStream(os);
+
+        for (ArchiveFile dataFile : dataFiles) {
+            ZipEntry entry = new ZipEntry(dataFile.getFilename());
+
+            zos.putNextEntry(entry);
+            zos.write(dataFile.getByteSource().read());
+            zos.closeEntry();
+        }
+
+
+        ZipEntry infoFileEntry = new ZipEntry(ARCHIVE_INFO_FILE_NAME);
+        zos.putNextEntry(infoFileEntry);
+        zos.write(RestUtils.GSON.toJson(archiveInfo).getBytes());
+        zos.closeEntry();
+        zos.close();
+        return zos;
+    }
+
+    public static class Builder implements WithBridgeConfig {
```

Comment:
:) In this case it makes it harder to use incorrectly, e.g. set the fields for surveys and non-survey activities and make an ambiguous data file